addons/account/models/account_move.py

```diff
@@ -1422,10 +1423,11 @@ def _fetch_duplicate_supplier_reference(self, only_posted=False):
 
         move_table_and_alias = "account_move AS move"
         place_holders = {}
-        if not moves.ids:
-            # This handles the special case of a record creation in the UI which isn't searchable in the DB
+        if not moves[0].id:  # check if record is under creation/edition in UI
+            # New record aren't searchable in the DB and record in edition aren't up to date yet
+            # Replace the table by safely injecting the values in the query
             place_holders = {
-                "id": 0,
+                "id": moves._origin.id or 0,
```

Comment:
There should only be one move because of this line above:
`if not moves[0].id: # check if record is under creation/edition in UI`

If we're in the UI, we should only have one move and if the assumption is right, it shouldn't crash.

It's a bit odd but I would prefer to keep the actual version so it would crash if we have more than one move.
That way, we would get a ticket and we would be aware about it.
If there are more than one, then, the code is bugged and we have to change several stuff in it to make it work that way.

Reply:
Seems like a good plan indeed !